### ISSUE TITLE
Using `--force-stderr-spec` option with `%PROGRAM_BASE%` `%TIME` and `%PID%`  generates a file name with garbage characters.

### ISSUE THREAD

**Issue by sebdelsol** [Nuitka/Nuitka#2572]
Using `--force-stderr-spec` option with `%PROGRAM_BASE%` `%TIME` and `%PID%`  generates a file name with garbage characters.
~~It worked fine with Nuitka < 1.9.~~

## Minimal sample
Create an empty `bug.py` file and compile it:
```console
python -m nuitka --force-stderr-spec="%PROGRAM_BASE% - %TIME% - %PID%.log" bug.py
```
Running `bug.cmd` creates a log file name with garbage characters: 
```console
bug - e133453008067015480 - 翽18412.log
bug - e133453008106092881 - 翽10748.log
```
## Nuitka 
**Environment:**
```console
1.9
Commercial: None
Python: 3.11.6 (tags/v3.11.6:8b6ee5b, Oct  2 2023, 14:57:12) [MSC v.1935 64 bit (AMD64)]
Flavor: CPython Official
Executable: C:\Users\sebmo\VS_Workspaces\nuitka\.venv\Scripts\python.exe
OS: Windows
Arch: x86_64
WindowsRelease: 10
Version C compiler: C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.36.32532\bin\Hostx64\x64\cl.exe (cl 14.3).
```

**Log:**
```console
Nuitka-Options:INFO: Used command line options: --force-stderr-spec="%PROGRAM_BASE% - %TIME% - %PID%.log" bug.py
Nuitka-Options:WARNING: You did not specify to follow or include anything but main program. Check options and make sure that is intended.
Nuitka:INFO: Starting Python compilation with Nuitka '1.9' on Python '3.11' commercial grade 'not installed'.
Nuitka:INFO: Completed Python level compilation and optimization.
Nuitka:INFO: Generating source code for C backend compiler.
Nuitka:INFO: Running data composer tool for optimal constant value handling.
Nuitka:INFO: Running C compilation via Scons.
Nuitka-Scons:INFO: Backend C compiler: cl (cl 14.3).
Nuitka-Scons:INFO: Backend linking program with 6 files (no progress information available for this stage).
Nuitka-Scons:INFO: Compiled 6 C files using clcache with 6 cache hits and 0 cache misses.
Nuitka:INFO: Keeping build directory 'bug.build'.
Nuitka:INFO: Successfully created 'bug.exe'.
Nuitka:INFO: Execute it by launching 'bug.cmd', the batch file needs to set environment.
```

**Comment by kayhayen** [comment]
I actually saw it too on Linux too.

**Comment by sebdelsol** [comment]
oh you're right it's broken with Nuitka 1.8.6 too: I only stumbled upon it when bumping Nuitka to 1.9.

**Comment by sebdelsol** [comment]
It works for me with your last commit 7a8e1a1

**Comment by kayhayen** [comment]
Well for the feature as described, a hello world would have been good enough. This is part of the 1.9.1 release I made.

**Comment by sebdelsol** [comment]
I guess you mean 1.9.2: I don't see those fixes in 1.9.1

**Comment by kayhayen** [comment]
My bad, you are correct, it's going out today though, fixed in 1.9.2 indeed. Overlapping hotfixes are too much to handle it seems.
